scripts/exrrfs_save_fcst.sh

```diff
@@ -45,53 +33,47 @@ else
 fi
 workdir="${DATAROOT}${ensindexstr}/${RUN}_fcst_${cyc}"
 umbrelladir="${UMBRELLA_DATA}/${jobid}"
-
 #
 #  move history files when it is done to umbrella
-#
-#  move the first history file 
-timestr=$(date -d "${CDATE:0:8} ${CDATE:8:2}" +%Y-%m-%d_%H.%M.%S) 
-if [[ -s "${workdir}/diag.${timestr}.nc" ]] && [[ -s "${workdir}/history.${timestr}.nc" ]] ;then
-  mv ${workdir}/history.${timestr}.nc ${umbrelladir}/.
-  mv ${workdir}/diag.${timestr}.nc    ${umbrelladir}/.
-  first_file_age=$(date -r ${umbrelladir}/diag.${timestr}.nc +%s)
-else
-  echo "history files do not exist, something is wrong!"
-  exit 1
-fi
-#
 #  now check each until the last history is moved
 #
-max_wait_time_sec=1200  # wait for 20 minutes
-while [[ ${min_file_age} -le ${max_wait_time_sec} ]]; do
+history_all=$(seq 0 $((10#${history_interval})) $((10#${fcst_len_hrs_thiscyc} )) )
+restart_all=$(seq 0 $((10#${restart_interval})) $((10#${fcst_len_hrs_thiscyc} )) )
+fhr_all=(${history_all})
+num_fhrs=${#fhr_all[@]}
+
+for (( ii=0; ii<${num_fhrs}; ii=ii+1 )); do
 
-  sleep 60s
-  last_file_age=${first_file_age}
+# get forecast hour and string
+    fhr=${fhr_all[$ii]}
+    CDATEp=$($NDATE ${fhr} ${CDATE} )
+    timestr=$(date -d "${CDATEp:0:8} ${CDATEp:8:2}" +%Y-%m-%d_%H.%M.%S)
 
-  for fhr in ${history_all}; do
-    CDATEp=$($NDATE $((10#${fhr} )) ${CDATE} )
-    timestr=$(date -d "${CDATEp:0:8} ${CDATEp:8:2}" +%Y-%m-%d_%H.%M.%S) 
-    if [[ -s ${umbrelladir}/diag.${timestr}.nc ]]; then
-      file_age=$(date -r ${umbrelladir}/diag.${timestr}.nc +%s)
-      if (( ${last_file_age} <= ${file_age} )); then
-        last_file_age=${file_age}
+# decide the history files
+    history_file=${workdir}/history.${timestr}.nc
+    diag_file=${workdir}/diag.${timestr}.nc
+
+# wait for file available for 20 min
+    for (( j=0; j < 20; j=j+1)); do
+      if [[ -s ${diag_file} ]]; then
+        break
       fi
-    fi
-    if [[ -s ${workdir}/diag.${timestr}.nc ]]; then
+      sleep 60s
+    done
+#
+    if [[ -s ${diag_file} ]] && [[ -s ${history_file} ]]; then
       sleep 10s
-      mv ${workdir}/history.${timestr}.nc ${umbrelladir}/.
-      mv ${workdir}/diag.${timestr}.nc    ${umbrelladir}/.
-      if (( ${fhr} == ${fcst_len_hrs_thiscyc} )); then
-	echo "Moved the last history file diag.${timestr}.nc. ALL DONE"
-	exit 0
+      mv ${history_file} ${umbrelladir}/.
+      mv ${diag_file}    ${umbrelladir}/.
+      # save to com
+      if (( ${ii} == 1 )); then
+        ${cpreq} ${umbrelladir}/history.${timestr}.nc ${comoutdir}/.
+        ${cpreq} ${umbrelladir}/diag.${timestr}.nc ${comoutdir}/.
```

Comment:
Line 69-71: why do we save some history and diag files into COM? Should we save mpasout files instead?

Reply:
This is testing the function to save mpasout and maybe other files to COM directory. Will replace with MPASSOUT when it is available.